examples/testdata/path_matcher/envoy_config.json

```diff
@@ -491,34 +491,34 @@
                           },
                           {
                             "decorator": {
-                              "operation": "ingress Google_Autogenerated_Unrecognized_Post_Method_Call"
+                              "operation": "ingress ESPv2_Autogenerated_CORS_GetShelf"
                             },
                             "match": {
                               "headers": [
                                 {
-                                  "exactMatch": "POST",
+                                  "exactMatch": "OPTIONS",
                                   "name": ":method"
                                 }
                               ],
                               "safeRegex": {
                                 "googleRe2": {},
-                                "regex": "^/.*$"
+                                "regex": "^/libraries/[^\\/]+/shelves/[^\\/]+$"
                               }
                             },
                             "route": {
-                              "cluster": "backend-cluster-http-bookstore-abc9876-uc.a.run.app:443",
-                              "hostRewriteLiteral": "http-bookstore-abc9876-uc.a.run.app",
+                              "cluster": "backend-cluster-http-bookstore-edf123456-uc.a.run.app:443",
+                              "hostRewriteLiteral": "http-bookstore-edf123456-uc.a.run.app",
                               "timeout": "15s"
                             }
                           },
                           {
                             "decorator": {
-                              "operation": "ingress Google_Autogenerated_Unrecognized_Put_Method_Call"
+                              "operation": "ingress ESPv2_Autogenerated_CORS_Google_Autogenerated_Unrecognized_Get_Method_Call"
```

Comment:
This is good.  It shows that /** routes are after other normal routes.  and /** for OPTIONS are after.  This should work.
<pre>
Normal routes:
allow_all wildcard routes for GET, POST ...  except OPTIONS
OPTIONS for normal http rules
OPTIONS for allow_all wildcard
</pre>
I am just nervous, that could this be accidentally,  or this order is guaranteed  by api-compler

Reply:
Right now, the OpenAPI compiler guarantees the `Google_Autogenerated_Unrecognized_...` will show up last.

I understand why you are nervous though, a future change in the OpenAPI compiler could break this. I plan to add a test to the OpenAPI compiler to guarantee the autogenerated methods are always last. This should make it safe to depend upon.